src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -2010,7 +2010,8 @@ public final void clearBatch() throws SQLServerException {
         } catch (SQLException e) {
             // throw a BatchUpdateException with the given error message, and return null for the updateCounts.
             throw new BatchUpdateException(e.getMessage(), null, 0, null);
-        } catch (IllegalArgumentException e) {
+        }
+        catch (IllegalArgumentException | StringIndexOutOfBoundsException e) {
```

Comment:
yup

Reply:
Please move this change out of this PR as we already have another active PR to track useBulkCopyWithBatchInsert API changes. It will be easier to track changes there.